src/tasks/deploy-sync.js

```diff
@@ -28,10 +29,16 @@ gulp.task('deploy:sync-init', () => {
   const environment = config.environment.split(/\s*,\s*|\s+/)[0];
 
   const envObj = tkConfig[environment];
+  const valid_id = utils.validateThemeId(envObj.theme_id);
   let proxyTarget = `https://${envObj.store}`;
 
-  if (envObj.theme_id && (envObj.theme_id === parseInt(envObj.theme_id, 10))) {
-    proxyTarget += `?preview_theme_id=${envObj.theme_id}`;
+  if (!valid_id) {
+    messages.invalidThemeId(envObj.theme_id, environment);
+    process.exit();
+  }
+
+  if (valid_id && typeof(valid_id) === "number") {
```

Comment:
I'd like to give a helpful message for fat-finger mistakes like `"1234r"'

Also, currently _any_ string as `theme_id` is falling back to the production theme.  Screenshot: http://take.ms/6Rv1r.  So I need to be stricter than just testing whether the id is the string `"live"`.  Forcing it to be an integer seemed like the best method.